spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java

```diff
@@ -672,6 +672,11 @@ public void setDefaultTopic(String defaultTopic) {
 		 */
 		private Long ackTime;
 
+		/**
+		 * True if this should be batch listener.
+		 */
+		private Boolean batchListener;
```

Comment:
What is the advantage of setting this to `null` by default? If there is a reasonable default that isn't likely to change, I'd rather have a `boolean` with the default value set. That way you can discover in your IDE if that's enabled by default or not.